pkg/overlay/overlay.go

```diff
@@ -30,20 +30,38 @@ func TempDir(containerDir string, rootUID, rootGID int) (string, error) {
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to create the overlay tmpdir in %s directory", contentDir)
 	}
-	upperDir := filepath.Join(contentDir, "upper")
-	workDir := filepath.Join(contentDir, "work")
+
+	return generateOverlayStructure(contentDir, rootUID, rootGID)
+}
+
+// ContainerOverlayRootfsDir generates an overlay directory structure for container content
```

Comment:
You are going to create this directory with overlay.ContainerOverlayRootfsDir,  Why not call this
```
func GenerateStructure(containerDir, ContainerID, name string, uid, gid int) (string, error){
 	contentDir := filepath.Join(containerDir, "overlay-containers", containerID, name)
        if err := idtools.MkdirAllAs(contentDir, 0700, uid, gid); err != nil {
		return "", errors.Wrapf(err, "failed to create the overlay %s directory", contentDir)
	}

	return generateOverlayStructure(contentDir, uid, gid)
}
 
Then call overlay.GenerateStructure(containerDir, ContainerID, "rootfs", uid, gid)

Reply:
@rhatdan done.